lib/benchee/statistics/percentile.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Benchee.Statistics.Percentile do
+  @moduledoc false
+
+  @doc """
+  Calculates the value at the `percentile_number`-th percentile. Think of this as the
+  value below which `percentile_number` percent of the samples lie. For example,
+  if `Benchee.Statistics.Percentile.percentile(samples, 99)` == 123.45,
+  99% of samples are less than 123.45.
```

Comment:
less than or equal to imo?

Reply:
Definitely open to improvements in the documentation. My understanding is that this value we are calculating is a "less than" number. Here's [what Wikipedia says](https://en.wikipedia.org/wiki/Percentile):

> A percentile (or a centile) is a measure used in statistics indicating the value below which a given percentage of observations in a group of observations fall. For example, the 20th percentile is the value (or score) below which 20% of the observations may be found.
>
> The term percentile and the related term percentile rank are often used in the reporting of scores from norm-referenced tests. For example, if a score is at the 86th percentile, where 86 is the percentile rank, it is equal to the value below which 86% of the observations may be found (carefully contrast with in the 86th percentile, which means the score is at or below the value of which 86% of the observations may be found - every score is in the 100th percentile).

Reply:
imo it's or equal. E.g.

When we have 10 values 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 - the 80th percentile would be 8 and 8 is a member of that 80th percentile group so from my understanding it should be less than or equal

Reply:
However....

```elixir
iex(2)> Percentile.percentile([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 80)
8.8
```

Reply:
but.... but... I guess I have to take a look at the calculation again then and the involved values or the algorithm used to obtain them. I had some suspicions about indexes being off somewhere... for my understanding of 80th percentile (aka 80 percent of the data) this should most definitely return 8 - or where am I going wrong?

Reply:
After cross checking this seems to be the right value (tm) but it still appearsto be weird to me. I mean 80% of my values are up until including 8. In this particular case I would have understood if it's like 8.2 or something... 8.8 just plain seems too weird. Need to up my percentile knowledge apparently